File: frontend/src/types/rag-dashboard.ts
/**
 * TypeScript types for RAG Dashboard
 *
 * Re-exports shared types from @qontinui/schemas and adds frontend-specific types.
 */

// JobStatus is a string-literal union type (Rust schemars → TS), not a
// runtime enum. Grep verified no `JobStatus.X` value uses exist, so
// type-only re-export is safe.
export type { JobStatus } from "@qontinui/shared-types/rag";

// Re-export types (must use 'export type' with isolatedModules)
export type {
  // Dashboard Stats
  JobSummary,
  RAGDashboardStats,
  // Embeddings
  EmbeddingItem,
  EmbeddingListResponse,
  // Jobs
  JobItem,
  JobListResponse,
  // Search
  SemanticSearchRequest,
  SearchResultItem,
  SemanticSearchResponse,
  // States
  StateFilterItem,
  StatesResponse,
} from "@qontinui/shared-types/rag";

// ============================================================================
// Frontend-specific types (not in shared schema)
// ============================================================================

export interface EmbeddingsParams {
  page?: number;
  limit?: number;
  state_filter?: string;
}

export interface JobsParams {
  page?: number;
  limit?: number;
  status_filter?: string;
}

File: backend/app/schemas/rag_dashboard.py
"""Schemas for RAG Dashboard endpoints.

Re-exports shared types from qontinui-schemas.
"""

# Re-export all RAG API schemas from the shared package
from qontinui_schemas.api.rag import (  # Enums; Dashboard Stats; Embeddings; Jobs; Search; States
    EmbeddingItem,
    EmbeddingListResponse,
    JobItem,
    JobListResponse,
    JobStatus,
    JobSummary,
    RAGDashboardStats,
    SearchResultItem,
    SemanticSearchRequest,
    SemanticSearchResponse,
    StateFilterItem,
    StatesResponse,
)

__all__ = [
    "JobStatus",
    "JobSummary",
    "RAGDashboardStats",
    "EmbeddingItem",
    "EmbeddingListResponse",
    "JobItem",
    "JobListResponse",
    "SemanticSearchRequest",
    "SearchResultItem",
    "SemanticSearchResponse",
    "StateFilterItem",
    "StatesResponse",
]
